apps/aens/src/aens.erl

```diff
@@ -0,0 +1,78 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2018, Aeternity Anstalt
+%%% @doc
+%%%    Resolve registered names
+%%% @end
+%%%-------------------------------------------------------------------
+
+-module(aens).
+
+%% API
+-export([resolve/2,
+         get_name_entry/1]).
+
+%%%===================================================================
+%%% Types
+%%%===================================================================
+
+-define(LABEL_SEPARATOR, <<".">>).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+-spec resolve(atom(), binary()) -> {ok, binary()} | {error, atom()}.
+resolve(Type, Binary) ->
+    case lists:reverse(binary:split(Binary, ?LABEL_SEPARATOR, [global, trim])) of
+        [Binary] ->
+            aec_base58c:safe_decode(Type, Binary);
+        [RegistrarNamespace|_Namespace] ->
+            %% When we have more registrars, pick faster and still gov friendly structure
+            %% If we allow to purchase registrars, this will live in regular tree
+            case [RN || RN <- aec_governance:name_registrars(), RN =:= RegistrarNamespace] of
+                [] ->
+                    {error, registrar_unknown};
+                _ ->
+                    {ok, StateTrees} = aec_conductor:top_state_trees(),
+                    NameTree = aec_trees:ns(StateTrees),
+                    case get_name(Binary, NameTree) of
+                        {ok, #{<<"pointers">> := Pointers}} ->
+                            case proplists:get_value(Type, Pointers) of
```

Comment:
I see this point relies on `Pointers` having atoms as keys - though this point could be adapted easily.